### PR TITLE
reduce line count using getattr() on layers 'weighted_sum...' rather than two explicit calls

### PR DESCRIPTION
**Pull request by ariejdl**

### CODE REVIEW

**Comment by ariejdl** [comment]
I signed it!

**Review by teamdandelion — CHANGES_REQUESTED** [review]
Rather than creating a string and using getattr, why not create a variable that has a reference to the correct function?

ie:

_function_ = layers.joint_weighted_sum_form_feature_columns if joint_weights else layers.weighted_sum_from_feature_columns

logits, _, _ = _function_(....)

(Although don't call it _function_ :P)

**Comment by ariejdl** [comment]
Thanks, yes, that's clearer

File: tensorflow/contrib/learn/python/learn/estimators/linear.py [hunk 1]
```diff
@@ -148,22 +148,16 @@ def _linear_model_fn(features, labels, mode, params, config=None):
...
-    else:
-      logits, _, _ = (
-          layers.weighted_sum_from_feature_columns(
-              columns_to_tensors=features,
-              feature_columns=feature_columns,
-              num_outputs=head.logits_dimension,
-              weight_collections=[parent_scope],
-              scope=scope))
+    
+    layer_fn = layers.joint_weighted_sum_from_feature_columns if joint_weights \
```
Comment (dandelionmane) [inline tensorflow/contrib/learn/python/learn/estimators/linear.py]: per the [google python style guide](https://google.github.io/styleguide/pyguide.html?showone=Conditional_Expressions#Conditional_Expressions) since this is longer than a 1 liner could you change it to a regular if statement?

**Review by teamdandelion — CHANGES_REQUESTED** [review]

**Review by teamdandelion — DISMISSED** [review]
oh man, sorry to keep nit-picking, but I think the pattern:

```
if x:
  a = b
else:
  a = c
```

is more clear than:
```
a = c
if x:
  a = b
```

File: tensorflow/contrib/learn/python/learn/estimators/linear.py [hunk 2]
```diff
@@ -99,7 +99,7 @@ def _add_bias_column(feature_columns, columns_to_tensors, bias_variable,
   columns_to_variables[bias_column] = [bias_variable]
 
 
-def _linear_model_fn(features, labels, mode, params, config=None):
+  def _linear_model_fn(features, labels, mode, params, config=None):
```
Comment (jhseu) [inline tensorflow/contrib/learn/python/learn/estimators/linear.py]: This shouldn't be indented

File: tensorflow/contrib/learn/python/learn/estimators/linear.py [hunk 3]
```diff
@@ -148,22 +148,19 @@ def _linear_model_fn(features, labels, mode, params, config=None):
       parent_scope,
       values=tuple(six.itervalues(features)),
       partitioner=partitioner) as scope:
+
```
Comment (jhseu) [inline tensorflow/contrib/learn/python/learn/estimators/linear.py]: Why extra empty lines? delete?

**Comment by ariejdl** [comment]
Thank you @jhseu think I've made the changes :)

**Review by jhseu — APPROVED** [review]

**Comment by jhseu** [comment]
Jenkins, test this please
